File: implement-shell-tools/ls/ls.js
const fs = require("fs");

const args = process.argv.slice(2);

const showAll = args.includes("-a");

// remove -a from arguments
const filteredArgs = args.filter(arg => arg !== "-a");

const target = filteredArgs[0] || ".";

let files = fs.readdirSync(target);

// hide hidden files if -a is NOT used
if (!showAll) {
  files = files.filter(file => !file.startsWith("."));
}

files.forEach(file => {
  console.log(file);
});

File: implement-shell-tools/ls/ls.py
import os
import sys

args = sys.argv[1:]

show_all = "-a" in args
one_per_line = "-1" in args

paths = [arg for arg in args if arg not in ["-a", "-1"]]

if not paths:
    paths = ["."]

for path in paths:
    items = os.listdir(path)

    if show_all:
        items.sort()
        items = [".", ".."] + items
    else:
        items = [item for item in items if not item.startswith(".")]
        items.sort()

    if one_per_line:
        for item in items:
            print(item)
    else:
        print(" ".join(items))